mlir/include/mlir/Dialect/MIOpen/MIOpenOps.td

```diff
@@ -150,28 +177,38 @@ def MIOpen_SubviewOp:
     The memref must be on either LDS (address space 3) or private (address
     space 5).
   }];
+  let assemblyFormat = [{
+    `(` $input `,` $offset `)` attr-dict `:` type($input) `to` type($output)
+  }];
 }
 
 // TBD: eventually replace this with linalg.fill?
 def MIOpen_FillOp:
-    MIOpen_Op<"fill">,
+    MIOpen_Op<"fill", [AllElementTypesMatch<["input", "value"]>]>,
     Arguments<(ins AnyMemRef:$input,
                    AnyTypeOf<[AnyInteger, AnyFloat]>:$value)> {
   let summary = "Fill memory with constant value on GPU";
   let description = [{
     The `miopen.fill` op fills a memref on GPU with a constant value.
   }];
+  let assemblyFormat = [{
+    `(` operands `)` attr-dict `:` type(operands)
+  }];
 }
 
 def MIOpen_MovePosV2Op:
-    MIOpen_Op<"move_pos_v2">,
+    MIOpen_Op<"move_pos_v2", [SameOperandsElementType,
+      AllTypesMatch<["input", "output"]>]>,
     Arguments<(ins VectorOfRankAndType<[1], [AnyInteger, AnyFloat]>:$input,
                    Variadic<AnyTypeOf<[AnyInteger, AnyFloat]>>:$values)>,
     Results<(outs VectorOfRankAndType<[1], [AnyInteger, AnyFloat]>:$output)> {
     let summary = "Add values to indices within the vector.";
     let description = [{
       The `miopen.move_pos_v2` op adds values to indices within the vector.
     }];
+    let assemblyFormat = [{
+      `(` operands `)` attr-dict `:` type($input) `,` type($values)
```

Comment:
same as above, looking at the updated unit test, I feel the syntax be a bit redundant to me.
for a vector of `i32` we should expect the offsets also be of type `i32`.

Reply:
Same equality issue as above, as well as the fact that the vector type is less constrained than the offsets.